### ISSUE TITLE
Bug in Accumulator with fixed capacity

### ISSUE THREAD

**Issue by dilawar** [SubconsciousCompute/SimpleAccumulator#4]
Consider this case where I make an accumulator of fixed size 3. 

```Rust
#[test]
fn test_sanity() {
    const CAPACITY : usize = 3;
    let mut acc = SimpleAccumulator::with_fixed_capacity::<f64>(&vec![], CAPACITY, true);

    let data = vec![0.0, 1.1, 2.2, 3.3, 4.4];
    for &v in &data {
        acc.push(v);
    }
    println!("{:?}", acc);
    assert_eq!(acc.vec.len(), 3);
    assert_eq!(acc.vec, vec![3.3,4.4,2.2]);
}
```

```
SimpleAccumulator { vec: [0.0, 3.3, 4.4], mean: 2.566666666666667, population_variance: 3.495555555555556, min: 0.0, min_: 0.0, max: 4.4, max_: 1.1, median: 2.3833333333333337, len: 3, capacity: 3, fixed_capacity: true, last_write_position: 2, accumulate: true }
thread 'test_sanity' panicked at 'assertion failed: `(left == right)`
  left: `[0.0, 3.3, 4.4]`,
 right: `[3.3, 4.4, 2.2]`', tests\test_accumulator_fixed.rs:16:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
test test_sanity ... FAILED```

**Comment by sn99** [comment]
On it

**Comment by dilawar** [comment]
This was fixed in #4. There may be others.

**Comment by sn99** [comment]
Waiting for VS studio/build to finish downloading, got LAN cable setup today in hostel room, 25-35mbps nice!
